cleanlab/outlier.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Class to identify which examples are out of distribution.
+"""
+import warnings
+from typing import Optional
+import numpy as np
+from cleanlab.rank import get_ood_scores, get_outlier_scores
+
+
+class OutOfDistribution:
+    OUTLIER_PARAMS = tuple(("k", "t"))
+    OOD_PARAMS = tuple(("labels", "adjust_pred_probs", "method"))
+    DEFAULT_PARAM_DICT = {
+        "k": None,  # outlier param
+        "t": 1,  # outlier param
+        "adjust_pred_probs": True,  # ood param
+        "method": "entropy",  # ood param
+    }
+
+    def __init__(self):
+        self.params = {}
+        self.knn = None
+        self.confident_thresholds = None
+
+    def fit_score(
+        self,
+        *,
+        features: Optional[np.ndarray] = None,
+        pred_probs: Optional[np.ndarray] = None,
+        labels: Optional[np.ndarray] = None,
+        params: dict = None,
+        verbose: bool = True,
+    ):
+
+        scores = self._shared_fit(
+            features=features, pred_probs=pred_probs, labels=labels, params=params
+        )  # some of below code belongs in def of _shared_fit() instead
+
+        if features is not None:
+            # fit to outlier scores
+            if verbose:
+                print("Computing outlier scores based on provided features ...")
+
+        if pred_probs is not None:
+            # fit to ood scores
+            if verbose:
+                print("Computing ood scores based on provided pred_probs ...")
```

Comment:
as your comment mentions, why not move this into _shared_fit ?

Reply:
Shared fit is called for fit() and fit_score(). Saying "Computing outlier scores based on provided features ..." does not make sense for fit function since we are not returning scores. 

Changing this to "Fitting OOD object based on provided features ..." and moving into _shared_fit() but let me know if I should revert